### ISSUE TITLE
wireless %essid and %quality displaying as question marks

### ISSUE THREAD

**Issue by SnifflyWhale** [i3/i3status#91]
As of version 2.10 my status line no longer telling me the SSID or the signal quality of my wireless connection, instead two question marks are displayed. My IP address is still displaying correctly. The line from my i3status config file:
wireless wlan0 {
        format_up = "Wifi: (%quality at %essid) %ip"
        format_down = "Wifi: down"
}
is showing up as:
Wifi: (? at ?) 192.168.1.13

**Comment by stapelberg** [comment]
We’ve switched from libiw to using libnl in commit 668bf2079f30e5df8f78974ebee2194d873f50ac, I guess that’s the immediate reason.

Which wireless driver are you using, and which Linux kernel version?

**Comment by SnifflyWhale** [comment]
I'm not sure about the driver but the device shows up in lsusb as:
Realtek Semiconductor Corp. RTL8192CU 802.11n WLAN Adapter
The Linux kernel version is 3.19.0-43-generic

**Comment by stapelberg** [comment]
Does upgrading to Linux 4.4 help? It contains the [new rtl8xxxu driver](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=26f1fad29ad973b0fb26a9ca3dcb2a73dde781aa) which uses the mac80211 stack instead of Realtek’s proprietary stack.

**Comment by core-1** [comment]
I second this issue.

A bit of info:

I'm using the **rtl8812au driver (https://github.com/gnab/rtl8812au)**

partial output:
```
$ ip link
3: enp0s20u10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode 
```
```
$ iwconfig
enp0s20u10  IEEE 802.11bgn  ESSID:"foobarbaz"  Nickname:"<WIFI@REALTEK>"
```

```
order += "cpu_temperature 0"
order += "wireless enp0s20u10"
...
...
$ sudo less /etc/i3status.conf
wireless enp0s20u10 {
        format_up = "W: (%essid:%quality:%ip)"
        format_down = "W: down"
}
```

Please note that ip does infact show up, but essid, quality, bitrate, and frequency do not.

Status line looks like this:
`W: (?:?:0 kb/s:?:192.168.1.70)`

Kernel: x86_64 Linux 4.3.3-2-ARCH

Because this is my primary system, I don't want to take risk trying to fix and recompile this, but at first glance it seems related to the aforementioned commit (I update frequently).

commit - 668bf20
cr - http://cr.i3wm.org/patch/692/

Hope this helps the assignee. Let me know if anything other info is needed. Cheers :)

**Comment by stapelberg** [comment]
@core-1 If you don’t think the commit is the reason, can you please use `git bisect` to identify the commit which broke the feature?

To be clear, in case it actually is our switch away from libiw, there’s nothing we can do about it. libnl is what drivers should use, and if any individual out-of-tree driver is not updated yet, that’s not a reason to jump through hoops in applications such as i3status.

**Comment by core-1** [comment]
Ahh nevermind. That was a guess. Upon verification (checkout commits/compile/run/revert/etc):

The detection works properly with commit 51c1dbdba28e9368 which is right before the commit you mentioned.

**Comment by core-1** [comment]
I guess as a workaround, I can always just replace the binary with a working one, but that's unnecessary. I think I'll just live with it..for now. I'll file a request on their github for now. Thanks! :)

**Comment by hyqneuron** [comment]
Can you guys at least give an option at make time for whether to use libiw or libnl? Right now the only thing I can do is going back to 51c1dbd. Also, the makefile should probably contain an option not to build the manpage. The dependency (on a kinda fresh Ubuntu 16.04) for it is about 1GB.

**Comment by eplanet** [comment]
@hyqneuron : for the Makefile issue, it is actually already possible for you to build i3status without the manpage since there are both directives _i3status_ and _manpage_.
```bash
# Only make i3status executable
make i3status
```
I agree however that you cannot install i3status without its manpage, but that definitely seems a good practice to me.

**Comment by stapelberg** [comment]
> Can you guys at least give an option at make time for whether to use libiw or libnl?

No. We want to use a single wireless library and fix the bugs instead of supporting two different libraries which work only for a subset of users.

> Also, the makefile should probably contain an option not to build the manpage. The dependency (on a kinda fresh Ubuntu 16.04) for it is about 1GB.

It sounds like you have recommends and suggests enabled in apt. If you turn them off, the dependencies set will become much smaller. Either way, we strongly recommend using distribution packages over building from source yourself.

**Comment by Fogapod** [comment]
Any updates on this? I'm having the exact same issue.

**Comment by mercuriete** [comment]
I have the same problem but with kernel driver:

kernel
```
uname -a
Linux portatil 5.4.38-gentoo #1 PREEMPT Thu Jun 4 12:21:36 WEST 2020 i686 Intel(R) Celeron(R) M processor 1.60GHz GenuineIntel GNU/Linux
```

device
```
lsusb | grep RTL
Bus 001 Device 002: ID 0bda:8179 Realtek Semiconductor Corp. RTL8188EUS 802.11n Wireless Network Adapter
```

driver
```
lsmod 
Module                  Size  Used by
r8188eu               360448  -2
lib80211               16384  -2
```

i3status version: ```x11-misc/i3status-2.12```

libnl version: ```dev-libs/libnl-3.5.0```

I will try to update my system that is a little bit outdated.

```

**Comment by Jandro20** [comment]
> wireless

Searching more info about i3status, I found this issue and I can do my bit with:

You can use the keyword "_first_" instead of search the wireless name to obtain the first name of your wireless list:

```
order += "wireless _first_"

wireless _first_ {
    format_up = "Wifi up (IP: %ip)"
    format_down = "" 
}
```

**Comment by nunotexbsd** [comment]
At FreeBSD I use `format_up = "W: (%quality at %essid, %bitrate) %ip"` and it shows `?` on `%quality` (the rest shows ok).

What the meaning of `%quality` and how should I dig to get it work on FreeBSD?
